src/api/v1/__tests__/api.json.test.unit.ts

```diff
@@ -0,0 +1,60 @@
+const apiJson = require('../api.json');
+import { identity as id } from 'ramda';
+
+const flatRoutes = Object.entries(apiJson.routes)
+  .reduce((acc, [resource, routes]) => {
+    const flatRoutes = Object.entries(routes)
+      .reduce((acc2, [endpoint, routeObjectsWithMethod]) => {
+        const flatRouteObjectsWithMedod = Object.entries(routeObjectsWithMethod)
+          .reduce((acc3, [method, routeObject]) => {
+            const flatRoute = {
+              path: `${resource}${endpoint}`,
+              method: `${method}`,
+              route: routeObject,
+            };
+            return [...acc3, flatRoute];
+          }, []);
+        return [...acc2, ...flatRouteObjectsWithMedod];
+      }, []);
+    return [...acc, ...flatRoutes];
+  }, []);
+
+const rx = new RegExp('[-:]');
+
+describe('Api.json structure', () => {
+  flatRoutes.forEach(({ path, method, route }) => {
+    describe(`::${method} ${path}`, () => {
+      const requestType = (method === 'GET')
+        ? 'query'
+        : 'body';
+      test('correct keys', () => {
+        expect(Object.keys(route)).toEqual([
+          'description',
+          'isImplemented',
+          'auth',
+          'intendedFor',
+          'scope',
+          requestType,
+          'response']);
+      });
+
+      test('correct values', () => {
+        expect(typeof route.description).toEqual('string');
```

Comment:
What's the advantage of doing it like this instead of using Joi schema?

Reply:
I think of `joi` as runtime validation, rather than part of a testing suite. But I guess I can write a schema and validate it during a test? Do we have any examples of using like this in the repo?

Reply:
Yeah, there's the config tests